### PR TITLE
Move node commands to top level

### LINKED ISSUE #926: [FEATURE] - cardano-cli compatibility command
In an effort to unnecessarily maintain support for prior eras (outside of testing) we are introducing a new command to `cardano-cli` that will replace the top level era commands (`shelley`, `allegra`, `alonzo`, `babbage`):

```
cabal run cardano-cli -- compatible                                                                                                      

Resolving dependencies...
Usage: cardano-cli compatible 
                                ( shelley
                                | allegra
                                | mary
                                | alonzo
                                | babbage
                                | conway
                                )

  Limited backward compatible commands for testing only.

Available options:
  -h,--help                Show this help text

Available commands:
  shelley                  Shelley era commands
  allegra                  Allegra era commands
  mary                     Mary era commands
  alonzo                   Alonzo era commands
  babbage                  Babbage era commands
  conway                   Conway era commands
```

Users are only interested in mainnet's era and the potential upcoming era. However our QA team has a test that hardforks from `Byron` to the latest era. Therefore there is some backwards compatibility we must maintain. The `compatible` command is intended to maintain those commands as required by our QA team. For all other use cases, users are advised to use preexisting commands.

This is the list of commands that QA uses (given by @mkoura )
- query tip
- ~byron genesis genesis~
- genesis create
- ~byron genesis print-genesis-hash~
- genesis hash
- ~node key-gen-KES~
- ~node issue-op-cert~
- ~byron key keygen~
- ~byron key signing-key-address~
- ~byron transaction issue-genesis-utxo-expenditure~
- ~key convert-byron-key~
- ~key verification-key~
- ~address build~
- ~address key-gen~
- stake-address key-gen
- ~address build~
- stake-address build
- stake-address registration-certificate
- ~node key-gen~
- stake-address delegation-certificate
- ~node issue-op-cert~
- stake-pool metadata-hash
- stake-pool registration-certificate
- ~byron transaction submit-tx~
- ~governance create-update-proposal~
- query protocol-parameters
- query utxo
- ~transaction build-raw~
- ~transaction sign~
- transaction submit

I will periodically cross off commands that have been integrated into the compatibility command in cardano-cli or have been moved to the top level as they are truly era agnostic. The `byron` command group can remain where it is for now.

### PR DESCRIPTION
**Pull request by Jimbo4350**
Partly resolves: #926
# Changelog

```yaml
- description: |
    The node commands are era agnostic and should be moved to the top level
# uncomment types applicable to the change:
  type:
  - feature        # introduces a new featur
  - compatible     # the API has changed but is non-breaking
```

# Context

Additional context for the PR goes here. If the PR fixes a particular issue please provide a [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=) to the issue.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Commands/Node.hs
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/EraBased/Commands/Node.hs:1]: In this case they should be moved out of `EraBased` module I think. Same for `Options` and `Run`.

Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/EraBased/Commands/Node.hs:1]: Good point

File: cardano-cli/src/Cardano/CLI/Options.hs [hunk 1]
```diff
@@ -47,13 +50,21 @@ pref =
       , helpRenderHelp customRenderHelp
       ]
 
+-- The node related commands are shelley era agnostic for the time being.
+-- There is no need to guard them by the era argument.
+nodeCmdsTopLevel :: Parser ClientCommand
+nodeCmdsTopLevel =
+  AnyEraCommand . AnyEraCommandOf ShelleyBasedEraShelley . NodeCmds <$> pNodeCmds
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/Options.hs]: I think I'm not getting the full context of this change.
1. Why we're hardcoding a witness here?
2. "for the time being" - are we planning to make `node` command era based again?
3. If the witness isn't needed, why not do it like `ping` or `debug`?

Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Options.hs]: 1. No longer applies
2. It's possible some requirement is added to running a node in the future. If that requirement requires a hardfork and is exposed as a cli command under the node group of commands, the node group of commands is no longer era agnostic.
3. No longer applies

**Review by smelc — APPROVED** [review]

**Comment by smelc** [comment]
@Jimbo4350> how about moving `debug check-node-configuration` (i.e. https://github.com/IntersectMBO/cardano-cli/pull/923) to `node check-configuration`?

**Comment by Jimbo4350** [comment]
> @Jimbo4350> how about moving `debug check-node-configuration` (i.e. #923) to `node check-configuration`?

Sure. In a follow up PR.

**Pull request merged**